truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -383,6 +383,12 @@ def copy_into_build_dir(from_path: Path, path_in_build_dir: str):
                     )
                 )
 
+        # User specified build commands
+        build_commands: list = []
+        if self._spec.build_commands is not None:
```

Comment:
you can just make the default an empty list and avoid having to do this all together